dbuf-core/src/location.rs

```diff
@@ -0,0 +1,34 @@
+//! Locations for parsed AST.
+
+/// Poosition in a document.
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Default)]
+pub struct Position {
+    /// Zero-based line position in a document.
+    pub line: u32,
+    /// Zero-based character position in a line.
+    pub character: u32,
+}
+
+impl Position {
+    pub fn new(line: u32, character: u32) -> Position {
+        Position { line, character }
+    }
+}
+
+/// Location in a document.
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Default)]
+pub struct Location {
+    /// Zero-based start of location.
```

Comment:
I guess, "zero-based" is a bit repetitive here since `Position`s were already described as zero-based